src/common/options.h

```diff
@@ -239,6 +256,12 @@ struct Option {
     return *this;
   }
 
+  Option& set_unit(const unit_t u)
```

Comment:
Hmm. But wouldn't that still involve setting some fields in `Option`? Also, do you think this would be generic enough, in case we need some units in future? 
@jcsp thoughts?

Reply:
I think it's simpler to add the time separately, to avoid having to convert lots of existing settings in the various places they're used.  It's also simpler if we keep a 1:1 relation between our type_t and the underlying POD type: if we added a bytes type then we would end up with two different types that were both an underlying integer.

Reply:
> to avoid having to convert lots of existing settings in the various places they're used.

we don't need to do so, if only we offer an implicit operator (uint64_t) for the size type.

> It's also simpler if we keep a 1:1 relation between our type_t and the underlying POD type:

true. a plain integer type and a "size" type.

by introducing the unit field instead of specializing the size type, we increase the size of every option -- we could avoid doing so.